Microsoft.Toolkit.Uwp.UI.Animations/Xaml/ImplicitAnimationSet.cs

```diff
@@ -6,18 +6,81 @@
+    /// Adding incompatible elements cannot be validated at build-time, but will result in a runtime crash.
+    /// </para>
+    /// <para>
+    /// Animations will monitor for changes in real-time to any of their public properties. For instance, if a binding is used to dynamically update the
+    /// <see cref="Animation{TValue, TKeyFrame}.To"/> or <see cref="Animation{TValue, TKeyFrame}.From"/> properties, the entire animation set will be
+    /// initialized again and assigned to the underlying <see cref="Visual"/> object for the targeted <see cref="UIElement"/>. This does not currently apply
+    /// to changes to the <see cref="Animation{TValue, TKeyFrame}.KeyFrames"/> property though (other than the entire property being reassigned). To achieve
+    /// dynamic updates to animation sets in that case, either leverage expression keyframes or just use code-behind to manually reinitialize the animations.
+    /// </para>
     /// </summary>
+    /// <remarks>
+    /// An <see cref="ImplicitAnimationSet"/> instance can only be used on a single <see cref="UIElement"/> target, and it cannot be shared across multiple
+    /// elements. Attempting to do so will result in a runtime crash. Furthermore, it is recommended not to move <see cref="IImplicitTimeline"/> instances from
+    /// one <see cref="ImplicitAnimationSet"/> to another, and doing so will add unnecessary runtime overhead over time. If you want to apply the same animations
+    /// to multiple elements, simply create another <see cref="ImplicitAnimationSet"/> instance and another set of animations with the same properties within it.
+    /// </remarks>
     public sealed class ImplicitAnimationSet : DependencyObjectCollection
     {
+        /// <summary>
+        /// Raised whenever any configuration change occurrs within the current <see cref="ImplicitAnimationSet"/> instance.
+        /// </summary>
+        internal event EventHandler? AnimationsChanged;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ImplicitAnimationSet"/> class.
+        /// </summary>
+        public ImplicitAnimationSet()
+        {
+            VectorChanged += ImplicitAnimationSetVectorChanged;
+        }
+
+        /// <summary>
+        /// Registers <see cref="RaiseAnimationsChanged(object, EventArgs)"/> for every added animation.
+        /// </summary>
+        /// <param name="sender">The current vector of animations.</param>
+        /// <param name="event">The <see cref="IVectorChangedEventArgs"/> instance for the current event.</param>
+        private void ImplicitAnimationSetVectorChanged(IObservableVector<DependencyObject> sender, IVectorChangedEventArgs @event)
+        {
+            if (@event.CollectionChange == CollectionChange.ItemInserted ||
+                @event.CollectionChange == CollectionChange.ItemChanged)
+            {
+                IInternalImplicitAnimation item = (IInternalImplicitAnimation)sender[(int)@event.Index];
```

Comment:
Do we accept/support anything other than ImplicitAnimation in this collection? I beliebe we do, so we should do a safe cast here.

Reply:
We don't we only support implicit animations here that are defined through the `IImplicitTimeline` interface. In practice we also technically allow users to just add whatever `DependencyObject` too since we're forced to use `DependencyObjectCollection`, which doesn't let us restrict the type of contained items, but we document not to do that so I think a runtime crash in that case would be acceptable. Though now that you made me think about this, and also to reply to your other question below, what we could do would be to remove the internal interface and just add the event to the public `IImplicitTimeline` interface.

**Pro**: users writing custom impliciti animations could support real-time updates as well
**Cons**: it's a breaking change (adding a member to a public interface)

Otherwise I could just make this a safe cast and just bail if it's not that interface, and we could document that if consumers write their own custom implicit animations, real-time updates will no longer be supported. This would avoid a breaking change, but it's also less limited (and possibly less intuitive for users?). I personally think taking a breaking change here could be acceptable as I don't think anyone would've added custom animations already after less than 2 weeks since the release? 🤔

Thoughts?

Reply:
I'm ok with this "breaking change" of adding the member to the public interface, but we shouldn't do this on a non-major release, and unfortunately it is already public. @michael-hawker what do you think? Should we keep the new interface and just make this a safe cast?

Reply:
Is there a reason we couldn't add the new event to the `IImplicitTimeline` interface itself? Then we get both the pro and the con, eh? ImplicitAnimation is the only implementer, right?

I'm fine with either approach, even though 7.0.1 is a patch, we're still trying to fix things that we discovered in 7.0.0. Like I'll delist 7.0.0 once we ship 7.0.1.

Reply:
We could, the thing is that `IImplicitTimeline` is a public interface which is supported for custom animations in a set, and adding members to public interfaces is a breaking change. As in, if someone is using 7.0 and had a custom animation they defined, with a new member in that interface updating to 7.0.1 would be breaking for them, as they'd have to add that event first to be able to compile.

So it's really just a matter of whether it's fine to take this minor breaking change now. Reasonably speaking, I'd be surprised if anyone had already defined custom animations this past week since 7.0 was released. Also technically fixing that would just be a matter of adding that empty event, so there's that too 🙂

Reply:
Update: I made the breaking change in https://github.com/windows-toolkit/WindowsCommunityToolkit/pull/3843/commits/4bc18c7b15000f99a068afe8a5cf193db0cf7613 👍